Filter out duplicate LineComments

Vynce opened the pull request:
Exact duplicate `LineComment`s don't make sense, so add an `equals()` method to the `LineComment` class that compares the line number and message and change `CommentedFile` to store a Set of unique `LineComments` instead of a List. A new `CommentedFile(String fileName)` constructor can be used to create an empty `CommentedFile` object which can then be populated using the `addLineComment()` method.

I also removed the unused `Comments` class and updated the minimum Java version to 7 to use the `Objects.equals()` method. I can rework `LineComment.equals()` if maintaining Java 6 support really is necessary.

rsandell commented on src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/rest/Comments.java:
```diff
@@ -1,54 +0,0 @@
- * A list of {@link LineComment}.
- * Not used.
- */
-public class Comments {
```
Hmm, though I like your design of putting the line comments inside the commented file object better than what it was before, I don't like removing classes and breaking binary compatibility.
Is there some way we can keep this class around but deprecate it and let ReviewInput still accept it as a deprecated way? It would make me sleep better :)

rsandell commented on src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/rest/LineComment.java:
```diff
@@ -44,4 +46,36 @@ public LineComment(int line, String message) {
+        }
+
+        LineComment e = (LineComment)o;
+        return Objects.equals(line, e.line)
```
Though I'm fine with bumping to java 7 _(we could even go up to 8 at this point)_ doing it just to be able to use `Objects` in a PR seems quite overkill when modern IDEs usually can generate a perfectly fine `equals` and `hashCode` method without it for you ;).
You are welcome to file a separate PR that bumps to Java 7 _(or 8)_ for better changelog tracking.

Vynce commented:
Thanks for taking a look! Those are both good points and I've removed the Java 7 change and added the `Comments` class back in this commit.

I can't find any use of the `Comments` class in gerrit-events or gerrit-trigger, but I suppose there may be other users of it. Let me know if you'd like me to replace its `List<LineComment>` with `Set<LineComment>`, similar to `ReviewInput`, or mark it as deprecated or anything -- either way it doesn't have any impact on my use case.

Vynce commented:
@rsandell, ping :smile:

Vynce commented:
@rsandell, could you please take another look at this change when you get a chance?

rsandell commented:
Sorry for the delay, have a lot of other things on my plate atm.

Vynce commented:
Thanks! No worries :smile:
